python/pyarrow/tests/test_dataset.py

```diff
@@ -334,7 +333,8 @@ def test_dataset_execute_iterator(dataset):
     # ARROW-11596: this would segfault due to Cython raising
     # StopIteration without holding the GIL. (Fixed on Cython master,
     # post 3.0a6)
-    tasks = dataset.scan()
+    with pytest.deprecated_call():
```

Comment:
At the very least we should create a JIRA to migrate these over to the new scan.  Although I'm wondering if we want to just do that now because that would give us a lot more coverage of the non-deprecated path.

Reply:
I'll do it now.